domain/src/test/java/org/oppia/domain/profile/ProfileManagementControllerTest.kt

```diff
@@ -49,14 +57,28 @@ class ProfileManagementControllerTest {
   @JvmField
   val mockitoRule: MockitoRule = MockitoJUnit.rule()
 
+  @Inject
+  lateinit var context: Context
+
   @Inject
   lateinit var profileManagementController: ProfileManagementController
 
   @Mock
-  lateinit var mockProfileManagementObserver: Observer<AsyncResult<Any?>>
+  lateinit var mockProfilesObserver: Observer<AsyncResult<List<Profile>>>
+  @Captor
+  lateinit var profilesResultCaptor: ArgumentCaptor<AsyncResult<List<Profile>>>
+
+  @Mock
+  lateinit var mockProfileObserver: Observer<AsyncResult<Profile>>
+  @Captor
+  lateinit var profileResultCaptor: ArgumentCaptor<AsyncResult<Profile>>
 
+  @Mock
+  lateinit var mockUpdateResultObserver: Observer<AsyncResult<Any?>>
   @Captor
-  lateinit var profileManagementResultCaptor: ArgumentCaptor<AsyncResult<Any?>>
+  lateinit var updateResultCaptor: ArgumentCaptor<AsyncResult<Any?>>
+
+  private val profilesList = mutableListOf<Profile>()
```

Comment:
Why make this mutable? We could make it an effective constant:

```kotlin
private val PROFILES_LIST = listOf(Profile.newBuilder()...build(), Profile.newBuilder()...)
```

Reply:
Updated to a constant listOf profiles.